### ISSUE TITLE
Doesn't build with gccgo

### ISSUE THREAD

**Issue by felixonmars** [mongodb/mongo-tools#18]
I tried to build mongo-tools and got the following errors with gccgo 5.1.0, which claims to have a complete implementation of go 1.4.2.

```
# github.com/spacemonkeygo/openssl
In file included from $WORK/github.com/spacemonkeygo/openssl/_obj/_cgo_main.c:12:0:
vendor/src/github.com/spacemonkeygo/openssl/ctx.go:74:12: error: 'verify_cb' redeclared as different kind of symbol
 extern int verify_cb(int ok, X509_STORE_CTX* store);
            ^
/tmp/go-build760887168/github.com/spacemonkeygo/openssl/_obj/_cgo_main.c:9:13: note: previous declaration of 'verify_cb' was here
 extern char verify_cb[];
             ^
In file included from $WORK/github.com/spacemonkeygo/openssl/_obj/_cgo_main.c:12:0:
vendor/src/github.com/spacemonkeygo/openssl/ctx.go:82:12: error: 'password_cb' redeclared as different kind of symbol
 extern int password_cb(char *buf, int size, int rwflag, void *password);
            ^
/tmp/go-build760887168/github.com/spacemonkeygo/openssl/_obj/_cgo_main.c:6:13: note: previous declaration of 'password_cb' was here
 extern char password_cb[];
             ^
```

Not sure about the problem since I am not familiar with Go, though. Hope it's still helpful!

**Comment by bryanpkc** [comment]
I am running into the same issue. The problem is probably fixed in go 1.5 but that doesn't help gccgo 5.1:

https://github.com/golang/go/issues/9294

**Comment by felixonmars** [comment]
Works now with gcc 5.3.0.
